examples/spouts/spout-marker/README.md

```diff
@@ -0,0 +1,328 @@
+# Resuming a Spout Pipeline
+
+Pachyderm enables you to create a special pipeline
+called *the spout* that enables you to ingest streaming
+data from an external source into Pachyderm. An example
+of such data could be a message queue, a transaction
+log, or others.
+
+Some of these streaming platforms can keep track of
+the current record so that in case of a network failure,
+the progress can be resumed from where it was left off.
+For example, Apache® Kafka tracks messages that
+will be sent to a Kafka consumer by recording the position of
+a pointer to the last record. This pointer is called an offset.
+If a Kafka consumer fails, it can then read the offset and resume
+from the position of the last processed message.
+
+When you use such a system in conjunction with Pachyderm,
+this progress needs to be tracked within Pachyderm as well.
+A spout has an option to specify a file or directory in
+which Pachyderm can keep track of the Kafka offsets or
+of similar record position trackers.
+This file is called a *spout marker* or just *marker*.
+
+A spout marker records the progress of a spout pipeline,
+and in case of an error, modification, or interruption
+can resume where it left off.
+
+In this example, we will create a spout pipeline with a
+marker file that will track the progress of a pipeline.
+Then, we will modify the pipeline and observe
+how the spout continues to update records without interruption.
+
+## Prerequisites
+
+Before you begin, verify that you have the following components
+installed on your machine:
+
+* Pachyderm 1.9.10 or later
+* Terminal
+
+## Pipeline Overview
+
+In this example, we will use a simple spout pipeline that
+will add dots into a spout marker file. Here is how the
+spout file will look like:
```

Comment:
I think calling it the `marker file` would be more clear, since the spout will likely make several normal files as part of its normal behavior.